wireshark: use relative cmake LIBDIR

B4dM4n opened the pull request:
###### Motivation for this change

`wireshark` expects `CMAKE_INSTALL_LIBDIR` to be relative and prefixes
it with `CMAKE_INSTALL_PREFIX` in some places to make it absolute.
This results in duplicate absolute paths being accessed, which don't exist like:
```
/nix/store/d1lshac7vwnmhl09fwc04pzmcgn1y219-wireshark-qt-3.2.2//nix/store/d1lshac7vwnmhl09fwc04pzmcgn1y219-wireshark-qt-3.2.2/lib/wireshark/extcap
```

Therefore `extcap` binaries like `sshdump` can't be used.

Making `CMAKE_INSTALL_LIBDIR` relative fixes this issue.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

bjornfor commented:
Is there an upstream issue? (Because this _is_ an upstream issue, since CMAKE_INSTALL_LIBDIR is allowed to be absolute.)

bjornfor reviewed (approved):
I'd like to see a reference to upstream issue though (it is their bug).

B4dM4n commented:
I couldn't find any upstream issue about this, but changes that involve `CMAKE_INSTALL_LIBDIR` tend to favor relative paths: [34574](https://code.wireshark.org/review/#/c/34574/) [23043](https://code.wireshark.org/review/#/c/23043/)

bjornfor commented:
That's unfortunate, since it's documented to allow absolute paths: "If the value is not already an absolute path, an absolute path is constructed typically by [...]." (https://cmake.org/cmake/help/v3.16/module/GNUInstallDirs.html).

B4dM4n committed:
wireshark: use relative cmake LIBDIR

`wireshark` expects `CMAKE_INSTALL_LIBDIR` to be relative and prefixes
it with `CMAKE_INSTALL_PREFIX` in some places to make it absolute.
This results in duplicate absolute paths being accessed, which don't exist.

Making `CMAKE_INSTALL_LIBDIR` relative fixes this issue.

B4dM4n commented:
I opened an upstream issue and embedded the link.

https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=16444

bjornfor reviewed (approved):
Great!
